File: bindings/wasm/examples/objects_by_type.mjs
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { GraphQlClient, ObjectFilter, initAsync } from "@iota/sdk-wasm";

await initAsync();

const client = GraphQlClient.newTestnet();

const coins = await client.objects(
  ObjectFilter.new({ typeTag: "0x2::coin::Coin<0x2::iota::IOTA>" }),
);

if (coins.data.length === 0) {
  console.log("No IOTA coin objects found");
} else {
  console.log("IOTA coin object IDs:");
  for (const coin of coins.data) {
    console.log(coin.id().toHex());
  }
}

File: crates/iota-sdk/examples/objects_by_type.rs
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use iota_sdk::graphql_client::{Client, error::Result, query_types::ObjectFilter};

#[tokio::main]
async fn main() -> Result<()> {
    let client = Client::new_testnet();

    let coins = client
        .objects(
            ObjectFilter {
                type_: "0x2::coin::Coin<0x2::iota::IOTA>".to_owned().into(),
                ..Default::default()
            },
            Default::default(),
        )
        .await?;

    if coins.data.is_empty() {
        println!("No IOTA coin objects found");
    } else {
        println!("IOTA coin object IDs:");
        for coin in coins.data {
            println!("{}", coin.id());
        }
    }

    Ok(())
}
